.github/workflows/unit_tests_gpu.yml

```diff
@@ -0,0 +1,48 @@
+name: Unit tests GPU
+
+on:
+  push:
+    branches: [main]
+  pull_request:
```

Comment:
We might not want the PR trigger here

Reply:
@riedgar-ms Does branches: [main] mean it'll only run when merged to main? brainstorming out loud, is there a way to manually trigger on PR instead of automatically? Another option is to have it run on a "staging" branch or something of the sort which we can merge to before merging with main. That way we can have a maintainer vet that the code isn't malicious, but still verify that it passes GPU tests before we put on main. Thoughts?

Reply:
`push ...main` is (AIUI) the trigger which will run on the actual merge. The `workflow_dispatch` trigger is supposed to add a 'Run Workflow' button on the Actions page. Having said that, for the plain 'Unit Test' build, the list of available branches I'm seeing are only those for this repo, not forks.

We could have an extra 'staging' branch, but that would be an extra thing to manage. And if the workflow were being automatically run there, there would still be the potential problem of people sneaking in Bitcoin miners.

Reply:
Whenever there's a new contributor, I believe that the builds won't be run until someone listed on the guidance repo clicks on a button, so there is that, @Harsha-Nori